apps/server/src/routers/api/v1/filesystem.rs

```diff
@@ -133,3 +103,36 @@ pub async fn list_directory(
 		page_size,
 	))))
 }
+
+fn get_os_start_path() -> PathBuf {
+	#[cfg(target_os = "windows")]
+	return Path::new("C:\\").into();
+	#[cfg(target_family = "unix")]
+	return Path::new("/").into();
+}
+
+fn read_and_filter_directory(
+	start_path: &Path,
+) -> Result<Vec<DirectoryListingFile>, std::io::Error> {
+	let listing = std::fs::read_dir(&start_path)?;
+
+	let files = listing
+		.filter_map(|res| res.ok())
+		.filter_map(|entry| filter_if_hidden(entry))
+		.map(|entry| DirectoryListingFile::from_pathbuf(entry.path()))
+		.collect();
+
+	Ok(files)
+}
+
+fn filter_if_hidden(entry: DirEntry) -> Option<DirEntry> {
+	let path = entry.path();
+	let stem = path.file_stem().unwrap_or_default();
+
+	// Remove hidden files starting with period
+	if stem.to_str().unwrap_or_default().starts_with('.') {
+		return None;
+	}
+
+	Some(entry)
+}
```

Comment:
Not important for this PR, but might be worth exporting `PathUtils` trait from `stump_core` so we don't maintain duplicate logic, e.g. `PathUtils::is_hidden_file`.